src/main/java/org/visola/cs/datastructure/CircularQueue.java

```diff
@@ -0,0 +1,87 @@
+package org.visola.cs.datastructure;
+
+public class CircularQueue<T> implements Queue<T> {
+
+  private final Object[] items;
+
+  private int head = -1;
+  private int tail = -1;
+
+  public CircularQueue(int size) {
+    this.items = new Object[size];
+  }
+
+  @Override
+  public boolean enQueue(T value) {
+    int newPos = nextPos(tail);
+    if (newPos == head) {
+      return false;
```

Comment:
Will `false` be returned when `size` is `0`?

`nextPos` will return `0`. `0` is not equal to `-1` so this line won't be invoked?

Reply:
Oh... I guess I shouldn't let people create a queue with size zero. 🤔